File: src/mpi/c/hello_world_mpi.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[]) {
    int     rank_id, ierr, num_ranks;
    double  start_time, wall_time;
    
    printf("C MPI minimal working example\n");
    
    ierr = MPI_Init(&argc, &argv);
    start_time = MPI_Wtime();
    ierr = MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);
    ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank_id);
    
    if (rank_id == 0) {
        printf("Number of available processors = %d.\n", num_ranks);
    }
    printf("\tProcess number %d branching off.\n", rank_id);
    
    wall_time = MPI_Wtime() - start_time;
    
    if (rank_id == 0) {
        printf("Elapsed wallclock time = %8.6fs.\n", wall_time);
    }
    
    MPI_Finalize();
    return 0;
}

File: src/mpi/f90/hello_world_mpi.f
C     MPI Fortran minimal working example  
      
      PROGRAM HELLO
      
      INCLUDE 'mpif.h'
      
      INTEGER TID, RANKS, IERR, STATUS(MPI_STATUS_SIZE)
      
      CALL MPI_INIT(IERR)
      
      CALL MPI_COMM_SIZE(MPI_COMM_WORLD, RANKS, IERR)
      CALL MPI_COMM_RANK(MPI_COMM_WORLD, TID, IERR)
      
      IF (TID .EQ. 0) THEN
            PRINT *, 'Fortran90 MPI minimal working example'
            PRINT *, 'Number of available processors =', RANKS
      END IF
      
      PRINT *, 'Process number ', TID, ' branching off.'
      
      CALL MPI_FINALIZE(IERR)
      
      END
